apps/kahoot-it-clone/src/app/login/login.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component } from '@angular/core';
+import {
+  FormBuilder,
+  FormGroup,
+  FormControl,
+  Validators
+} from '@angular/forms';
+import { AuthService } from '../auth-service/auth.service';
+
+@Component({
+  selector: 'kahoot-it-clone-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent {
+  loginForm: FormGroup;
+
+  constructor(private fb: FormBuilder, private authService: AuthService) {
+    this.loginForm = this.fb.group({
+      email: new FormControl('', Validators.required),
+      password: new FormControl('', Validators.required)
+    });
+  }
+
+  onSubmit() {
+    const email = this.loginForm.get('email').value;
+    const password = this.loginForm.get('password').value;
+
```

Comment:
This can be simplified to const value = this.loginForm.value; That will return an object that looks like {email: '', password: ''}